server/webapp/WEB-INF/rails.new/app/controllers/api_v4/current_user_controller.rb

```diff
@@ -0,0 +1,45 @@
+##########################################################################
+# Copyright 2017 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module ApiV4
+  class CurrentUserController < ApiV4::BaseController
+    include Api::UsersHelper
+
+    before_action :check_user_and_404
+    before_action :load_current_user
+
+    def show
+      render DEFAULT_FORMAT => ApiV1::UserRepresenter.new(@user_to_operate).to_hash(url_builder: self)
+    end
+
+    def update
+      result           = HttpLocalizedOperationResult.new
+      @user_to_operate = save_user(result, @user_to_operate)
+      if result.isSuccessful
+        render DEFAULT_FORMAT => ApiV1::UserRepresenter.new(@user_to_operate).to_hash(url_builder: self)
```

Comment:
Never use a representer from a different version in a controller :)

The preference is that everything under `ApiV1` should only refer to classes under `ApiV1`. Makes it easier to reason about compatibility with no exceptions around. This also means a lot of duplicate code, but we are good about deleting the code when an API is deprecated and removed.